pkgs/development/libraries/libint/default.nix

```diff
@@ -1,76 +1,114 @@
-    "--with-g12-opt-am=3"
-    "--with-g12dkh-max-am=5"
-    "--with-g12dkh-opt-am=3"
-    "--enable-contracted-ints"
-    "--enable-shared"
-   ] ++ lib.optional enableFMA "--enable-fma";
-
-  preConfigure = ''
-    ./autogen.sh
-  '';
-
-  postBuild = ''
-    # build the fortran interface file
-    cd export/fortran
-    make libint_f.o ENABLE_FORTRAN=yes
-    cd ../..
-  '';
-
-  postInstall = ''
-    cp export/fortran/libint_f.mod $out/include/
-  '';
-
   meta = with lib; {
     description = "Library for the evaluation of molecular integrals of many-body operators over Gaussian functions";
     homepage = "https://github.com/evaleev/libint";
     license = with licenses; [ lgpl3Only gpl3Only ];
-    maintainers = [ maintainers.markuskowa ];
+    maintainers = with maintainers; [ markuskowa sheepforce ];
     platforms = [ "x86_64-linux" ];
   };
-}
+
+  codeGen = stdenv.mkDerivation rec {
+    inherit pname version;
+
+    src = fetchFromGitHub {
+      owner = "evaleev";
+      repo = pname;
+      rev = "v${version}";
+      sha256 = "0pbc2j928jyffhdp4x5bkw68mqmx610qqhnb223vdzr0n2yj5y19";
+    };
+
+    patches = [ ./fix-paths.patch ];
+
+    nativeBuildInputs = [
+      autoconf
+      automake
+      libtool
+      gfortran
+      mpfr
+      python3
+      perl
+      gmpxx
+    ];
+
+    buildInputs = [ boost eigen ];
+
+    preConfigure = "./autogen.sh";
+
+    configureFlags = [
+      #"--help"
+      #"--enable-eri=2"
+      #"--enable-eri3=2"
+      #"--enable-eri2=2"
+      #"--with-eri-max-am=7,5,4"
+      #"--with-eri-opt-am=3"
+      #"--with-eri3-max-am=7"
+      #"--with-eri2-max-am=7"
+      #"--with-g12-max-am=5"
+      #"--with-g12-opt-am=3"
+      #"--with-g12dkh-max-am=5"
+      #"--with-g12dkh-opt-am=3"
+      #"--enable-contracted-ints"
+      "--enable-fortran"
+      "--enable-shared"
+      ##
+      "--with-eri-max-am=2"
+    ] ++ lib.optional enableFMA "--enable-fma";
+
+    makeFlags = [ "export" ];
+
+    installPhase = ''
+      mkdir -p $out
+      cp ${pname}-${version}.tgz $out/.
+    '';
+
+    enableParallelBuilding = true;
+
+    inherit meta;
+  };
+
+  codeComp = stdenv.mkDerivation rec {
+    inherit pname version;
+
+    src = "${codeGen}/${pname}-${version}.tgz";
+
+    nativeBuildInputs = [
+      gfortran
+      python3
+      cmake
+    ];
+
+    buildInputs = [ boost eigen ];
+
+    cmakeFlags = [
+      "-DENABLE_FORTRAN=ON"
+
+      # Default is just "double", but SSE2 is available on all x86_64 CPUs.
+      "-DLIBINT2_REALTYPE=${if enableAVX
+        then "libint2::simd::VectorAVXDouble"
+        else "libint2::simd::VectorSSEDouble"}"
+    ];
+
+    # Can only build in the source-tree. A lot of preprocessing magic fails otherwise.
+    configurePhase = ''
+      runHook preConfigure
+
+      cmake -S. -B. ${builtins.toString cmakeFlags}
+
+      runHook postConfigure
+    '';
+
+    # Parallel building messes up Fortran interface compilation
+    enableParallelBuilding = true;
```

Comment:
Why is it then enabled?

Reply:
Still figuring out what is going on there with Fortran. I thought it was the parallel building, but actually it is not. Must be some other difference between manual compilation in a nix-shell and with `nix-build`. Leftover from my debugging purposes. Sorry for the confusion.